tests/meta/demultiplex/test_validation.py

```diff
@@ -19,45 +19,45 @@ def test_validate_sample_fastq_with_valid_file():
     sample_fastq = Path("Sample_123/sample_L0002.fastq.gz")
 
     # WHEN validating the sample fastq file
-    validate_sample_fastq_file(sample_fastq)
+    is_valid = is_valid_sample_fastq_file(sample_fastq=sample_fastq, sample_internal_id="sample")
 
-    # THEN no exception should be raised
+    # THEN it should be valid
+    assert is_valid
 
 
 def test_validate_sample_fastq_without_sample_id_in_parent_directory_name():
-    # GIVEN a sample fastq file without a sample id in the parent directory name
-    sample_fastq = Path("sample_L0002.fastq.gz")
+    # GIVEN a sample fastq file without a sample id in the parent directory name or file name
+    sample_fastq = Path("L0002.fastq.gz")
 
     # WHEN validating the sample fastq file
-    # THEN a ValueError should be raised
-    with pytest.raises(
-        ValueError, match="Parent directory name of sample fastq must contain 'Sample_<sample_id>'."
-    ):
-        validate_sample_fastq_file(sample_fastq)
+    is_valid_fastq = is_valid_sample_fastq_file(
+        sample_fastq=sample_fastq, sample_internal_id="sample_id"
+    )
+
+    # THEN it should not be valid
+    assert not is_valid_fastq
 
 
 def test_validate_sample_fastq_without_lane_number():
```

Comment:
I think in this case a docstring could clarify that the Lane number is in the path, not in an attribute of the sample. For example:
```suggestion
def test_validate_sample_fastq_without_lane_number():
    """Test that a sample fastq file without the lane in the path fails validation."""
```
These are things that are obvious for the person writing the function but not instantly for the reviewer. A docstring like this would save review time. That is why I am against removing docstrings from test functions

Reply:
I would personally prefer a short test name and a docstring than a long test name with no docstring. I think in this case is not that clear but other tests definitely require a docstring.